src/controller.cpp

```diff
@@ -221,12 +242,12 @@ void Controller::setPlan(
   controller_state_.current_global_plan_index = 0;
 
   // find closest current position to global plan
-  double minimum_distance_to_path = 1e3;
-  double dist_to_segment = 0;
-  double iterative_dist_to_goal = 0.0;
+  auto minimum_distance_to_path = 1e3 * boost::units::si::square_meter;
```

Comment:
Can't this be declared as: `boost::units::si::square_meter minimum_distance_to_path = 1e3;` ?

Reply:
No, the reasoning is that you could use the incorrect unit by mistake. For example write a value in centimeters, inches, feet, or whatever. So Boost units requires you to explicitly state what unit of measurement to use for the literal.

You could of course introduce a helper function to reduce verbosity:

```cpp
square_meter_t to_square_meter(double value)
{
  return value * boost::units::si::square_meter;
}
```

Reply:
Aah, check. 

Amount * unit. Does make sense.

No need for the helper. This will be more readable with the typedefs I think.